### PR TITLE
Event stuff

### PR DESCRIPTION
**Pull request by SciathGH**

### CODE REVIEW

File: Server/src/main/kotlin/net/horizonsend/ion/server/listeners/bukkit/PlayerPickUpItemListener.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import org.bukkit.event.EventPriority
+import org.bukkit.event.Listener
+import org.bukkit.event.player.PlayerAttemptPickupItemEvent
+
+@Suppress("unused")
+class PlayerPickUpItemListener : Listener {
+	@EventHandler(priority = EventPriority.MONITOR)
+	fun onPlayerPickUpItem(event: PlayerAttemptPickupItemEvent){
+		ionCore {
+		if (event.item == CustomItems.MINERAL_TITANIUM.singleItem()){
```
Comment (Peter-Crawley) [inline Server/src/main/kotlin/net/horizonsend/ion/server/listeners/bukkit/PlayerPickUpItemListener.kt:17]: use a `when` block, also learn how indentation works

File: Common/src/main/kotlin/net/horizonsend/ion/common/annotations/IonCore.kt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package net.horizonsend.ion.common.annotations
+/**
+ * This annotation should be used when using any code from IonCore in anywhere but IonCore
+ * Anything with this annotation should be replaced with Ion native code.
```
Comment (Peter-Crawley) [inline Common/src/main/kotlin/net/horizonsend/ion/common/annotations/IonCore.kt:4]:
```suggestion
 * Code annotated with this should be replaced if practical to do so
```

File: Common/src/main/kotlin/net/horizonsend/ion/common/annotations/IonCore.kt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package net.horizonsend.ion.common.annotations
```
Comment (Peter-Crawley) [inline Common/src/main/kotlin/net/horizonsend/ion/common/annotations/IonCore.kt:1]:
```suggestion
package net.horizonsend.ion.server.annotations
```

**Review by Peter-Crawley — CHANGES_REQUESTED** [review]
Annotate the `ionCore` wrapper with `@IonCore`

File: Server/src/main/kotlin/net/horizonsend/ion/server/listeners/bukkit/PlayerPickUpItemListener.kt (hunk 1 above)
Comment (SciathGH) [reply Server/src/main/kotlin/net/horizonsend/ion/server/listeners/bukkit/PlayerPickUpItemListener.kt:17]: it autoindents but ok

File: Server/src/main/kotlin/net/horizonsend/ion/server/annotations/IonCore.kt [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package net.horizonsend.ion.server.annotations
+/**
+ * This annotation should be used when using any code from IonCore in anywhere but IonCore
+ * * Code annotated with this should be replaced if practical to do so.
```
Comment (Peter-Crawley) [inline Server/src/main/kotlin/net/horizonsend/ion/server/annotations/IonCore.kt:4]:
```suggestion
 * Code annotated with this should be replaced if practical to do so.
```

**Review by Peter-Crawley — CHANGES_REQUESTED** [review]

**Pull request merged**